### PR TITLE
Task/test chat

### PR DESCRIPTION
**Pull request by misocho**

### CODE REVIEW

File: backend/chat/models.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Message(TimeStampedModel):
     class Meta:
         verbose_name = _("Message")
         verbose_name_plural = _("Messages")
+        ordering = ["-id"]
```
Comment (jlorencelim) [inline backend/chat/models.py:25]: Make sure to do `makemigrations` as changing the class `Meta` affects migrations

File: backend/chat/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import pytest
```
Comment (jlorencelim) [inline backend/chat/tests/test_models.py:1]: We have already started using Django unittests and not pytest. To keep it consistent, let's just use Django's unittest

File: backend/chat/tests/test_viewsets.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        )
+
+        self.message = MessageFactory.create(thread=self.thread, contact=self.user)
+
+    def test_create_thread(self):
+        data = {"participants": [self.user.id, self.teacher_user.id]}
+        response = self.client.post(reverse("chat-v1:threads-list"), data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_list_tread_messages(self):
```
Comment (jlorencelim) [inline backend/chat/tests/test_viewsets.py:39]: fix typo. should be `thread`

File: backend/chat/api/v1/serializers.py [hunk 4]
```diff
@@ -22,3 +23,27 @@ class Meta:
             "id",
             "participants",
         )
+
+    def validate_participants(self, participants):
```
Comment (jlorencelim) [inline backend/chat/api/v1/serializers.py:27]: Add unit tests for this validation

File: backend/chat/api/v1/serializers.py [hunk 5]
```diff
@@ -22,3 +23,27 @@ class Meta:
...
+
+        if user not in participants:
+            raise serializers.ValidationError(
+                f"{user.id} should be a participant in this thread."
+            )
+
+        if len(participants) != 2:
+            raise serializers.ValidationError("A thread should have two users.")
+
+        for participant in participants:
```
Comment (jlorencelim) [inline backend/chat/api/v1/serializers.py:38]: You can just do a single query for these conditions

File: backend/Pipfile [hunk 6]
```diff
@@ -35,3 +35,5 @@ channels = "==2.4.0"
 channels-redis = "*"
 daphne = "*"
 django-cors-headers = "*"
+pytest-asyncio = "*"
```
Comment (jlorencelim) [inline backend/Pipfile:38]: We have already started using Django unittests and not pytest. To keep it consistent, let's just use Django's unittest

**Review by jlorencelim — CHANGES_REQUESTED** [review]

File: backend/chat/api/v1/serializers.py (hunk 5 above)
Comment (misocho) [reply backend/chat/api/v1/serializers.py:38]: Hi. I'm not sure how to achieve this. Could you please elaborate.

Comment (jlorencelim) [reply backend/chat/api/v1/serializers.py:38]: Something like this. 
```
if participants.filter(user_type__in=[User.USER_TYPE_STUDENT, User.USER_TYPE_PARENT]):
    raise PermissionDenied
```

**Review by jlorencelim — CHANGES_REQUESTED** [review]

File: backend/chat/api/v1/serializers.py [hunk 7]
```diff
@@ -22,3 +23,30 @@ class Meta:
...
+
+        if user not in participants:
+            raise serializers.ValidationError(
+                f"{user.id} should be a participant in this thread."
+            )
+
+        if len(participants) != 2:
+            raise serializers.ValidationError("A thread should have two users.")
+
+        participant_filter = filter(
```
Comment (jlorencelim) [inline backend/chat/api/v1/serializers.py:38]: If the solution can be done via 1 django query, then do it. Looping through a django query via python is a heavy transaction. Something like this
```
participants
    .filter(user_type__in=[User.USER_TYPE_STUDENT, User.USER_TYPE_PARENT])
    .exclude(id=participant.id)
```

**Review by jlorencelim — CHANGES_REQUESTED** [review]

File: backend/chat/api/v1/serializers.py (hunk 7 above)
Comment (misocho) [reply backend/chat/api/v1/serializers.py:38]: `participants` is not a django query. It's a list of user objects, the django query inbuilt functions won't work on it.

**Review by jlorencelim — APPROVED** [review]

**Pull request merged**